### PR TITLE
Record index

### LINKED ISSUE #1697: Single source for the type of numerical record index
Currently there are three different entities that reasons about the index type of numerical records; the parameter configuration, the distribution implementation and the actual record implementation. This responsibility should be kept one place.

I suggest we make the record index type in the record implementation public such that the two others can utilise it.

### PR DESCRIPTION
**Pull request by markusdregi**
**Issue**
Resolves #1697 

**Approach**
Introduce and make public record index in `ert3.data` and use it in config and stats modules.

### CODE REVIEW

**Review by ManInFez — APPROVED** [review]
LGTM :bacon:

**Comment by markusdregi** [comment]
Temporary merge freeze until libres is back home! Do not merge this PR in the meantime!

**Pull request merged**
